README/ReleaseNotes/v612/index.md

```diff
@@ -112,40 +112,46 @@ large TClonesArray where each element contains another small vector container.
+  - Add `DefineSlot`, a `Define` transformation that is aware of the multi-threading slot where the workload is executed
+  - Add `DefineSlotEntry`, a `Define` transformation that is aware of the multi-threading slot and of the current entry number
+  - Add `GetColumnsNames`: users can now get the names of the available columns coming from trees, data sources or `Define`d columns
   - Add `OnPartialResult` and `OnPartialResultSlot`: users can now register one or more functions to be executed on partial results of TDF actions during the event loop.
     This mechanism is meant to be used to inspect partial results of the analysis or print useful debug information.
     For example, both in single- and multi-thread event loops, one can draw a result histogram and update the canvas every 100 entries like this:
-    ```
+    ```c++
     auto h = tdf.Histo1D("x");
     TCanvas c("c","x hist");
     h.OnPartialResult(100, [&c](TH1D &h_) { c.cd(); h_.Draw(); c.Update(); });
     ```
     See the tutorials for more examples.
-  - Added `DefineSlotEntry`, a `Define` transformation that is aware of the multi-threading slot and of the current entry number
-  - Users can now access multi-threading slot and entry number as pre-defined columns "tdfslot_" and "tdfentry_"
-  - Added a `Sum` action that sums all values of a column for the processed entries
-  - Added support for friend trees in all cases
+  - Add `Sum`, an action that sums all values of a column for the processed entries
+  - The new TDataSource interface allows developers to pipe any kind of columnar data format into TDataFrame. Two example data sources have been provided: the TRootDS and the TTrivialDS. The former allows to read via the novel data source mechanism ROOT data, while the latter is a simple generator, created for testing and didactic purposes. It is therefore now possible to interface *any* kind of dataset/data format to ROOT as long as an adaptor which implements the pure virtual methods of the TDataSource interface can be written in C++.
+  - TDF can now read CSV files through a specialized TDataSource. Just create the TDF with `MakeCsvDataFrame("f.csv", false, ',')`
+  - Users can now configure Snapshot to use different file open modes ("RECREATE" or "UPDATE"), compression level, compression algorithm, TTree split-level and autoflush settings
+  - Users can now access multi-threading slot and entry number as pre-defined columns "tdfslot_" and "tdfentry_". Especially useful for pyROOT users.
+  - Users can now specify filters and definitions as strings containing multiple C++ expressions, e.g. "static int a = 0; return ++a". Especially useful for pyROOT users.
+  - pyROOT users can now easily specify parameters for the TDF histograms and profiles thanks to the newly introduced tuple-initialization
+  - Add support for friend trees
```

Comment:
Perhaps we can elaborate a bit more here? Proposal: "Add support for friend trees/chains. If you created a TDataFrame from a tree, now you can e.g. select and use its friends' columns from a `Filter` transformation or access them when you are `Define`ing a new column.

Reply:
👍 will add